syntax = "proto3";

service Humidifier {
    rpc change_smart_humidifier_state (ChangeSmartHumidifierRequest) returns (HumidifierResponse) {}
    rpc change_state (ChangeHumidifierStateRequest) returns (HumidifierResponse) {}
    rpc get_smart_humidifier_state (EmptyRequest) returns (HumidifierResponse) {}
    rpc get_state (EmptyRequest) returns (HumidifierResponse) {}
    rpc change_bounds(ChangeBoundsHumidifierRequest) returns (HumidifierResponse) {}
}

message EmptyRequest {}

message ChangeBoundsHumidifierRequest {
    optional int32 upper_bound = 1;
    optional int32 lower_bound = 2;
}

message ChangeSmartHumidifierRequest {
    optional bool state = 1;
}

message ChangeHumidifierStateRequest {
    optional bool state = 1;
}

message HumidifierResponse {
    optional bool status = 1;
    optional string message = 2;
}